src/classes/HH_HouseholdNaming.cls

```diff
@@ -218,6 +233,39 @@ public without sharing class HH_HouseholdNaming {
     }
 
     /*******************************************************************************************************
+    * @description Returns Contact fields specified in the Household Naming Settings format fields.
+    * This is a top layer for iNaming.setHouseholdNameFieldsOnContact() method which can be also implemented 
+    * by a client so it is important to validate returned Contact fields. 
+    * @return Set<String> If the Automatic Household Naming is enabled, return set of Contact field API names; 
+    * otherwise, an empty set
+    */
+    public Set<String> getHouseholdNamingContactFields() {
+        if (!isAutomaticNaming()) {
+            return new Set<String>();
+        }
+    
+        Set<String> fieldNames = iNaming.setHouseholdNameFieldsOnContact();
+        if (fieldNames == null || fieldNames.isEmpty()) {
+            return new Set<String>();
+        }
+
+        Map<String, Schema.DescribeFieldResult> contactFieldResultsByName = UTIL_Describe.getFieldResultsByName(Contact.SObjectType);
```

Comment:
The `isValidField(String objectName, String fieldName)` does not handle relationships. All new methods added in `Util_Describe` are there to resolve any case field name (firstname, FirstName) regardless if it is a field on an SObject or a relationship field.